code/framework/engine/src/main/java/io/cattle/platform/engine/eventing/impl/ProcessEventListenerImpl.java

```diff
@@ -81,6 +91,25 @@ public void processExecute(Event event) {
         }
     }
 
+    private boolean shouldWaitLonger(long processId) {
+        ProcessRecord record = processRecordDao.getRecord(processId);
+        if (record == null || record.getEndTime() != null) {
+            // is there any additional processing that happens or can we
+            // even short circuit it here and be done with this event?
+            // for now, assume additional processing occurs to get rid
+            // of event
+            return false;
+        }
+        int numPreviousAttempts = processRecordDao.getNumPreviousExecutions(processId);
+        if (numPreviousAttempts < 2) {
```

Comment:
I wanted to give some leeway for the first retry in case of slight differences in clock times